// @dasha.ai/platform-sdk/dist/tts.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TtsManager = void 0;
const querystring = __importStar(require("querystring"));
const axios_1 = __importDefault(require("axios"));
const errors_1 = require("./errors");
class TtsManager {
    constructor(options) {
        // TODO: optimize work with the Rest API #732
        this._client = axios_1.default.create({
            baseURL: (options.insecure ? "http://" : "https://") + options.server,
            headers: {
                Authorization: `Bearer ${options.apiKey}`,
                "Content-Type": "application/json",
            },
            paramsSerializer: querystring.stringify,
            responseType: "arraybuffer",
        });
        this._client.interceptors.response.use(undefined, (error) => {
            var _a, _b, _c, _d, _e, _f, _g;
            const validationErrors = (_b = (_a = error.response) === null || _a === void 0 ? void 0 : _a.data) === null || _b === void 0 ? void 0 : _b.errors;
            if (validationErrors) {
                const messages = Object.values(validationErrors).flat().join(" ");
                return Promise.reject(new errors_1.DashaSdkError(messages, error));
            }
            const errorDetails = (_d = (_c = error.response) === null || _c === void 0 ? void 0 : _c.data) === null || _d === void 0 ? void 0 : _d.details;
            if (errorDetails) {
                return Promise.reject(new errors_1.DashaSdkError(errorDetails, error));
            }
            const errorReason = (_f = (_e = error.response) === null || _e === void 0 ? void 0 : _e.data) === null || _f === void 0 ? void 0 : _f.reason;
            if (errorReason) {
                return Promise.reject(new errors_1.DashaSdkError(errorReason, error));
            }
            const statusText = (_g = error.response) === null || _g === void 0 ? void 0 : _g.statusText;
            if (statusText) {
                return Promise.reject(new errors_1.DashaSdkError(statusText, error));
            }
            return Promise.reject(new errors_1.DashaSdkError(error.message, error));
        });
    }
    async _postSynthesis(body) {
        return new Uint8Array((await this._client.post("/api/v1/tts/synthesis", body)).data);
    }
    async synthesize(text, options) {
        const { configName, voiceInfo } = options;
        const { speaker, language: lang, emotion, speed, variation } = voiceInfo;
        return await this._postSynthesis({
            text,
            configName: configName !== null && configName !== void 0 ? configName : "Default",
            voiceInfo: {
                speaker,
                lang,
                emotion,
                speed: speed !== null && speed !== void 0 ? speed : 1,
                variation: variation !== null && variation !== void 0 ? variation : 0,
            },
        });
    }
}
exports.TtsManager = TtsManager;
